File: app/src/main/java/io/catter2/login/LoginUseCase.java
package io.catter2.login;

import android.util.Log;

public class LoginUseCase {
    /**
     * @param username
     * @param password
     * @return A unique identifiable token representing user associated to the given username.
     * Will return null if username or password are invalid.
     */
    public String login(String username, String password) {
        Log.d("LoginUseCase", "Login: " + username + ", " + password);
        try {
            LoginService service = new LoginService();
            String token = service.login(username, password);
            Log.d("LoginUseCase", "Login token: " + token);
            return token;
        } catch (Exception e) {
            Log.e("LoginUseCase", "Login failed");
            return null;
        }
    }
}

File: app/src/main/java/io/catter2/login/LoginUseCase.kt
package io.catter2.login

import android.util.Log

class LoginUseCase {
    /**
     * @param username
     * @param password
     * @return A unique identifiable token representing user associated to the given username.
     * Will return null if username or password are invalid.
     */
    fun login(username: String, password: String): String? {
        Log.d("LoginUseCase", "Login: $username, $password")
        return try {
            val service = LoginService()
            val token = service.login(username, password)
            Log.d("LoginUseCase", "Login token: " + token)
            token
        } catch (e: Exception) {
            Log.e("LoginUseCase", "Login failed")
            null
        }

    }
}
